builders/interact/ethereum-api/libraries/viem.md

```diff
@@ -0,0 +1,324 @@
+
+### Send Transaction Script {: #send-transaction-script }
+
+You'll only need one file to execute a transaction between accounts. For this example, you'll be transferring 1 UNIT token from an origin address on the demo EVM ContainerChain (from which you hold the private key) to another address. To get started, you can create a `transaction.ts` file by running:
+
+```bash
+touch transaction.ts
+```
+
+Next, you will create the script for this file and complete the following steps:
+
+1. Update your imports to include `createPublicClient`, `createWalletClient`, `http`, `parseEther`, and `defineChain` functions from `viem`, as well as the `privateKeyToAccount` function from `viem/accounts`
+2. Define the chain details of your EVM ContainerChain, making sure to include all fields shown below. Both `public` and `default` RPC URLs are required to be listed, even if they are same
+3. [Set up a viem wallet client](#for-writing-chain-data) for writing chain data, which can be used along with your private key to send transactions. **Note: This is for example purposes only. Never store your private keys in a TypeScript file**
+4. [Set up a public viem client](#for-reading-chain-data) for reading chain data, which will be used to wait for the transaction receipt
+5. Define the `addressTo` variable
+6. Create the asynchronous `send` function, which wraps the transaction object and the `walletClient.sendTransaction` method
+7. Use the `walletClient.sendTransaction` function to sign and send the transaction. You'll need to pass in the transaction object, which only requires the recipient's address and the amount to send. Note that `parseEther` can be used, which handles the necessary unit conversions from Ether to Wei, similar to using `parseUnits(value, decimals)`. Use `await` to wait until the transaction is processed and the transaction hash is returned
+8. Use the `publicClient.waitForTransactionReceipt` function to wait for the transaction receipt, signaling that the transaction has been completed. This is particularly helpful if you need the transaction receipt or if you're running the `balances.ts` script directly after this one to check if the balances have been updated as expected
+9. Lastly, run the `send` function
+
+??? code "View transaction.ts"
+
+    ```ts
+    --8<-- 'code/ethereum-api/viem/transaction.ts'
+    ```
+
+To run the script, you can run the following command in your terminal:
+
+```bash
+npx ts-node transaction.ts
+```
+
+If the transaction was successful, in your terminal you'll see the transaction hash has been printed out. You can also use the `balances.ts` script to check that the balances for the origin and receiving accounts have changed. The entire workflow would look like this:
+
+![The result of running the transaction and balances scripts in the terminal](/images/builders/interact/ethereum-api/viem/viem-2.png)
+
+## Deploy a Contract {: #deploy-contract }
+
+--8<-- 'text/libraries/contract.md'
```

Comment:
This snippet has a grammar issue: "Next you can add the Solidity code to the file:" should be "Next, you..."